backend/server/src/services/tournamentService.ts

```diff
@@ -24,28 +25,40 @@ export class TournamentService {
     return await newTournament.toObject();
   }
 
-  // listing
-  // public async getAllTournaments(): Promise<Tournament[]> {
-  //     console.log("I am in get all tournament service")
-  //     return await TournamentModel.find().exec();
-  // }
-
-  // // all running
-  // public async getRunningTournaments(): Promise<Tournament[]> {
-  //     const currentDate = new Date();
-
-  //     return await TournamentModel.find({
-  //         startDate: { $lte: currentDate },
-  //         endDate: { $gte: currentDate }
-  //     }).exec();
-  // }
-
-  // // all future tournaments
-  // public async getUpcomingTournaments(): Promise<Tournament[]> {
-  //     const currentDate = new Date();
-
-  //     return await TournamentModel.find({
-  //         startDate: { $gt: currentDate }
-  //     }).exec();
-  // }
+  // update - add players
+  public async addPlayerToTournament(
+    tournamentId: string,
+    playerId: string
+  ): Promise<Tournament> {
+    const tournament = await TournamentModel.findById(tournamentId).exec();
+
+    if (tournament === null || tournament === undefined) {
+      throw new NotFoundError({
+        message: "Tournament not found"
+      });
+    }
+
+    // Check if player exists in the UserModel
+    const player = await UserModel.findById(playerId).exec();
+    if (player === null || player === undefined) {
+      throw new NotFoundError({
+        message: "Player not found"
+      });
+    }
+
+    // Check if the player is already in the tournament
+    if (tournament.players.includes(playerId)) {
+      throw new Error("Player already registered in the tournament");
+    }
+
+    // Check if the tournament has reached its maximum number of players
+    if (tournament.players.length >= tournament.maxPlayers) {
+      throw new Error("Tournament has reached its maximum number of players");
```

Comment:
I would throw a `BadRequestError` instead, since throwing regular `Error` would result in a internal error, which this is not.